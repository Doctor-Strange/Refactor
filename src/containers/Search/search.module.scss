.search_box {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.4rem;
  form {
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    .search_box_div {
      margin: auto 1rem;
      input {
        padding: 0.94rem;
      }
      .Search_Text_transparent {
        margin-bottom: 4px;
    color: transparent;
    cursor: default;
    line-height: 1;
      }
      display: inline-block;
      width: 100%;
      .DatePicker {
        width: 100%;
        input {
          text-align: right;
          padding: 0.94rem;
          font-family: inherit;
          margin: 0.3rem auto 0.5rem;
          display: block;
          width: 100%;
          outline: 0;
          -webkit-appearance: none;
          line-height: 1;
          font-size: 1.4rem;
          background: #fff;
          border: 1px solid rgba(34, 36, 38, 0.15);
          color: rgba(0, 0, 0, 0.87);
          border-radius: 0.4rem;
          box-shadow: 0 0 0 0 transparent inset;
        }
      }
    }
    .search_Btn {
      width: 100%;
    }
  }
}
