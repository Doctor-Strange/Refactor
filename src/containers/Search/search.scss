@import "../../styles/colors";
.search_box {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.4rem;
  form {
    width: 100%;
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    .search_box_div {
      margin: auto 1rem;
      display: inline-block;
      width: 25%;
      input {
        padding: 0.94rem;
        margin: 0.3rem auto;
      }
      .Search_Text_transparent {
        margin-bottom: 4px;
        color: transparent;
        cursor: default;
        line-height: 1;
      }
    }
    .Date_picker_container {
      position: relative;
      margin: auto 1rem;
      display: inline-block;
      width: 50%;
      .date_Input_Container {
        display: flex;
        .input_container {
          width: 100%;
          input {
            font-family: inherit;
            margin: 0.3rem auto;
            display: block;
            width: 100%;
            outline: 0;
            -webkit-appearance: none;
            line-height: 1.5;
            border: 1px solid $input_border;
            border-left: 0px solid $input_border;
            border-radius: 0.4rem;
            border-top-left-radius: 0px;
            border-bottom-left-radius: 0px;
            padding: 0.94rem;
            font-size: 1.4rem;
            background: #fff;
            color: rgba(0, 0, 0, 0.87);
            box-shadow: 0 0 0 0 transparent inset;
          }
          .exception_input {
            border: 1px solid $input_border;
            border-left: 1px solid $input_border;
            border-right: 1px solid $input_border;
            border-radius: 0.4rem;
            border-top-left-radius: 4px;
            border-top-right-radius: 0px;
            border-bottom-left-radius: 4px;
            border-bottom-right-radius: 0px;
          }
        }
        .DatePicker {
          position: absolute;
          width: 100%;
          height: 100%;
          input {
            cursor: pointer;
            opacity: 0;
            text-align: right;
            padding: 0.94rem;
            font-family: inherit;
            margin: 0.3rem auto 0.5rem;
            display: block;
            width: 100%;
            height: 100%;
            outline: 0;
            -webkit-appearance: none;
            line-height: 1;
            font-size: 1.4rem;
            background: #fff;
            border: 1px solid $input_border;
            color: rgba(0, 0, 0, 0.87);
            border-radius: 0.4rem;
            box-shadow: 0 0 0 0 transparent inset;
          }
        }
      }
    }
    .search_Btn {
      width: 100%;
    }
  }
}
@media screen and(max-width:820px) {
  .search_box_div {
    p {
      font-size: 1rem;
      line-height: 2.1;
    }
  }
}
@media screen and(max-width:720px) {
  .search_box {
    display: block;
    form {
      display: block;
      .Date_picker_container {
        width: 100%;
        margin: 1rem auto;
      }
      .search_box_div {
        margin: 1rem auto;
        width: 100%;
        p {
          font-size: 1.4rem;
        }
        input {
        }
        .Search_Text_transparent {
        }
        .DatePicker {
          input {
          }
        }
      }
      .search_Btn {
      }
    }
  }
}

@media screen and(max-width:480px) {
  .search_box {
    display: block;
    form {
      display: block;
      .Date_picker_container {
        width: 100%;
        margin: 0;
        // .date_Input_Container {
        //   margin: 0rem auto;
        //   display: block;
        //   .input_container {
        //     margin-bottom: 1rem;
        //     margin-bottom: 1rem;
        //     input{
        //       border: 1px solid $input_border;
        //       border-radius: 4px !important;
        //     }
        //   }
        // }
      }
    }
  }
}
