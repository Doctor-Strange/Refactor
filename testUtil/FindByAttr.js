

export const FindByAttr = (wrapper, value) => {
  return wrapper.find(`[data-test="${value}"]`);
};
