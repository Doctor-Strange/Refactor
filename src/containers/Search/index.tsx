import React, { useState, useEffect, useContext } from "react";

import "react-modern-calendar-datepicker/lib/DatePicker.css";
import DatePicker, { DayRange, utils } from "react-modern-calendar-datepicker";
import moment from "moment-jalaali";

import DropdownSearch from "../../components/form/Dropdown";
import { REQUEST_GET_LOCATION } from "../../API/index";

import Router from "next/router";

import "./search.scss";
import Button from "../../components/form/Button";

import modal_context from "../../context/Modal_context";

moment.loadPersian({ dialect: "persian-modern" });

const Search = (props: ISearch) => {
  const [LocationId, setLocationId] = useState(1);
  const [fromDay, setFromDay] = useState("");
  const [toDay, setToDay] = useState("");
  const [dayRange, setDayRange] = React.useState<DayRange>({
    from: null,
    to: null,
  });

  const [locationsList, setLocationsList] = useState([]);
  const [loading, setLoading] = useState(false);

  const MODAL_CONTEXT = useContext(modal_context);

  const get_car_location = async () => {
    const res: any = await REQUEST_GET_LOCATION();
    setLocationsList(res.data);
  };

  useEffect(() => {
    get_car_location();
    setDateFromStorage();
  }, []);

  const setDateFromStorage = () => {
    if (localStorage["start"] && localStorage["end"]) {
      if (localStorage["start"] !== null && localStorage["end"] !== null) {
        let start = JSON.parse(localStorage["start"]);
        if (start.day > moment().jDate()) {
          if (start.month >= moment().jMonth() + 1) {
            setDayRange({
              from: JSON.parse(localStorage["start"]),
              to: JSON.parse(localStorage["end"]),
            });
          } else {
            localStorage.removeItem("start");
            localStorage.removeItem("end");
          }
        } else if (start.month > moment().jMonth() + 1) {
          setDayRange({
            from: JSON.parse(localStorage["start"]),
            to: JSON.parse(localStorage["end"]),
          });
        } else {
          localStorage.removeItem("start");
          localStorage.removeItem("end");
        }
      }
    }
  };

  const GotoSearchResult = (e) => {
    e.preventDefault();
    if (dayRange.from && dayRange.to) {
      if (props.dynamic) {
        props.searchSubmit({
          location_id: 1,
          date: {
            Start_date: `${dayRange.from.year}/${dayRange.from.month}/${dayRange.from.day}`,
            End_date: `${dayRange.to.year}/${dayRange.to.month}/${dayRange.to.day}`,
          },
        });
        return;
      }
      localStorage["start"] = JSON.stringify(dayRange.from);
      localStorage["end"] = JSON.stringify(dayRange.to);
      setLoading(true);
      Router.push({
        pathname: "/search-result",
        query: {
          location_id: LocationId,
          start_date: `${dayRange.from.year}/${dayRange.from.month}/${dayRange.from.day}`,
          end_date: `${dayRange.to.year}/${dayRange.to.month}/${dayRange.to.day}`,
        },
      });
    }
  };

  useEffect(() => {
    if (dayRange.from) {
      setFromDay(convertDate(dayRange.from));
    } else {
      setFromDay(" ");
      setToDay(" ");
    }
    if (dayRange.to) {
      setToDay(convertDate(dayRange.to));
    } else {
      setToDay(" ");
    }
  }, [dayRange]);

  const convertDate = (v) => {
    let value = moment(`${v.year}/${v.month}/${v.day}`, "jYYYY/jM/jD").format(
      "dddd jDD jMMMM jYY"
    );
    return value;
  };

  return (
    <section className="search_box">
      <form data-test-id="GotoSearchResult" onSubmit={(e) => GotoSearchResult(e)}>
        <div className="search_box_div">
          <p className="label">خودرو خود را کجا تحویل میگیرید؟</p>
          <DropdownSearch
            data={locationsList}
            InputDisable={true}
            hardValue="تهران"
            Select={(i) => {
              // setLocationId(i.key);
              localStorage["User_Location"] = JSON.stringify(i);
              MODAL_CONTEXT.modalHandler("TellMe");
            }}
            clearField={() => setLocationId(1)}
          />
        </div>
        <div className="Date_picker_container">
          <div className="date_Input_Container">
            <DatePicker
              value={dayRange}
              onChange={setDayRange}
              shouldHighlightWeekends
              minimumDate={utils("fa").getToday()}
              locale="fa"
              colorPrimary="#4ba3ce"
              disabledDays={[utils("fa").getToday()]}
            />
            <div className="input_container">
              <p className="label">از تاریخ</p>
              <input readOnly={true} value={fromDay ? fromDay : ""}></input>
            </div>
            <div className="input_container">
              <p className="label">تا تاریخ</p>
              <input
                readOnly={true}
                className="exception_input"
                value={toDay ? toDay : ""}
              ></input>
            </div>
          </div>
        </div>
        <div className="search_box_div">
          <p className="Search_Text_transparent">search</p>
          <Button
            value="جستجو"
            class="Blue_BTN search_Btn"
            loading={loading}
            click={() => {}}
          />
        </div>
      </form>
    </section>
  );
};

interface ISearch {
  dynamic?: boolean;
  searchSubmit?: any;
}

export default Search;
