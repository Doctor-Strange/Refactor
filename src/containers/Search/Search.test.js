import React from "react";
import { shallow } from "enzyme";
import { FindByAttr } from "../../../testUtil/FindByAttr";
import Search from "./";

const searchSubmitMock = jest.fn();

const props = {
  dynamic: true,
  searchSubmit: searchSubmitMock,
};

const setup = (props) => {
  return shallow(<Search {...props} />);
};

describe("Search component", () => {
  describe("snapshot", () => {
    test("/components/Search", () => {
      const wrapper = setup(props);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
