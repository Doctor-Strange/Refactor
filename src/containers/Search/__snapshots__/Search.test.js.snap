// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component snapshot /components/Search 1`] = `ShallowWrapper {}`;
